File: semana_4_5_patrones_creacionales/codigo_ejemplos/ejemplo1_singleton.py
# Ejemplo 1: Singleton en Python (VERSIÓN CRUZADA del ejemplo C++)

class Config:
    """Versión Python del Singleton del PDF"""
    _instance = None
    
    def __new__(cls):
        if cls._instance is None:
            cls._instance = super().__new__(cls)
        return cls._instance
    
    def show_message(self):
        print("Configuración global cargada.")

# Prueba
obj1 = Config()
obj2 = Config()
obj1.show_message()
print(f"¿Son iguales? {obj1 is obj2}")

File: semana_4_5_patrones_creacionales/codigo_ejemplos/ejemplo1_singleton.cpp
// Ejemplo 1: Singleton en C++ (EXACTO DEL PDF)

#include <iostream>
using namespace std;

class Config {
private:
    static Config* instance;
    Config() {} // Constructor privado
public:
    static Config* getInstance() {
        if (!instance) instance = new Config();
        return instance;
    }
    void showMessage() { cout << "Configuración global cargada.\n"; }
};

Config* Config::instance = nullptr;

int main() {
    Config* obj1 = Config::getInstance();
    Config* obj2 = Config::getInstance();
    obj1->showMessage();
    cout << "¿Son iguales? " << (obj1 == obj2) << endl;
}
